classes/gateways/class.pmprogateway_stripe.php

```diff
@@ -4300,55 +4303,53 @@ public function refund( &$order, $transaction_id = null ) {
 			}
 		}
 
-		//get the charge
+		$success = false;
+
+		//attempt refund
 		try {
-			$charge = Stripe_Charge::retrieve( $transaction_id );
-		} catch ( \Throwable $e ) {
-			$charge = false;
-		} catch ( \Exception $e ) {
-			$charge = false;
-		}
+			
+			$secretkey = pmpro_getOption( "stripe_secretkey" );
 
-		//can't find the charge?
-		if ( empty( $charge ) ) {
-			$order->status     = "error";
-			$order->errorcode  = "";
-			$order->error      = "";
-			$order->shorterror = "";
+			// If they are not using legacy keys, get Stripe Connect keys for the relevant environment.
+			if ( ! self::using_legacy_keys() && empty( $secretkey ) ) {
+				if ( pmpro_getOption( 'gateway_environment' ) === 'live' ) {
+					$secretkey = pmpro_getOption( 'live_stripe_connect_secretkey' );
+				} else {
+					$secretkey = pmpro_getOption( 'sandbox_stripe_connect_secretkey' );;
+				}
+			} 
 
-			return false;
-		}
+			$client = new Stripe_Client( $secretkey );
+			$refund = $client->refunds->create( [
+				'charge' => $transaction_id,
+			] );
 
-		//attempt refund
-		try {
-			$refund = $charge->refund();
-		} catch ( \Throwable $e ) {
-			$order->errorcode  = true;
-			$order->error      = __( "Error: ", 'paid-memberships-pro' ) . $e->getMessage();
-			$order->shorterror = $order->error;
+			$success = true;
+			
+			global $current_user;
 
-			return false;
-		} catch ( \Exception $e ) {
-			$order->errorcode  = true;
-			$order->error      = __( "Error: ", 'paid-memberships-pro' ) . $e->getMessage();
-			$order->shorterror = $order->error;
+			$notes = $order->notes;
+			$order->notes = $notes.' '.sprintf( __('Order successfully refunded on %1s for transation ID %2s by %3s', 'paid-memberships-pro' ), date_i18n('Y-m-d H:i:s'), $transaction_id, $current_user->display_name );	
```

Comment:
Should we separate with `\n\n` and `trim()` the notes?

Should we add a function that does this for a string more easily or not worth it?

Reply:
Missing translator note for the placeholders

Reply:
This should use the proper position syntax like `%1$s`

```suggestion
			$order->notes = $notes.' '.sprintf( __('Order successfully refunded on %1$s for transation ID %2$s by %3$s', 'paid-memberships-pro' ), date_i18n('Y-m-d H:i:s'), $transaction_id, $current_user->display_name );	
```

Reply:
should be transaction, not translation